oak-lucene/src/test/java/org/apache/jackrabbit/oak/plugins/index/lucene/dynamicBoost/DynamicBoostTest.java

```diff
@@ -77,100 +75,142 @@ protected void createTestIndexNode() throws Exception {
-        NodeTypeRegistry.register(root, toInputStream(ASSET_NODE_TYPE), "test nodeType");
-        Tree props = createIndex("dam:Asset", true);
-        Tree pt = createNodeWithType(props, "predictedTags", UNSTRUCTURED);
-        pt.setProperty("name", "jcr:content/metadata/predictedTags/.*");
-        pt.setProperty("isRegexp", true);
-        pt.setProperty("dynamicBoost", true);
-        pt.setProperty("propertyIndex", true);
-        root.commit();
+    @Test
+    public void withDynamicBoostLite() throws Exception {
+        createDynamicBoostIndex(true);
 
-        String log = runTest(LuceneDocumentMaker.class, true);
+        String log = runIndexingTest(LuceneDocumentMaker.class, true);
         assertEquals("[]", log);
     }
 
-    @Test public void withDynamicBoostMissingProperty() throws Exception {
-        NodeTypeRegistry.register(root, toInputStream(ASSET_NODE_TYPE), "test nodeType");
-        Tree props = createIndex("dam:Asset");
-        Tree pt = createNodeWithType(props, "predictedTags", UNSTRUCTURED);
-        pt.setProperty("name", "jcr:content/metadata/predictedTags/.*");
-        pt.setProperty("isRegexp", true);
-        pt.setProperty("dynamicBoost", true);
-        pt.setProperty("propertyIndex", true);
-        root.commit();
+    @Test
+    public void withDynamicBoostMissingProperty() throws Exception {
+        createDynamicBoostIndex(true);
 
-        String log = runTest(LuceneDocumentMaker.class, false);
+        String log = runIndexingTest(LuceneDocumentMaker.class, false);
         assertEquals("[]", log);
     }
 
-    private String runTest(Class<?> loggerClass, boolean nameProperty) throws CommitFailedException {
-        LogCustomizer customLogs = LogCustomizer
-                .forLogger(loggerClass)
-                .enable(Level.TRACE).create();
+    @Test
+    public void testQueryDynamicBoost() throws Exception {
+        createDynamicBoostIndex(false);
+
+        factory.indexFieldProvider = new IndexFieldProviderImpl();
+        factory.queryTermsProvider = new FulltextQueryTermsProviderImpl();
+
+        Tree test = createNodeWithType(root.getTree("/"), "test", UNSTRUCTURED);
+
+        Tree predicted1 = createAssetNode(test, "item1");
+        createPredictedTag(predicted1, "red", 0.1);
+        predicted1.setProperty("x", "red");
+
+        Tree predicted2 = createAssetNode(test, "item2");
+        createPredictedTag(predicted2, "red", 0.9);
+        predicted2.setProperty("x", "red");
+
+        Tree predicted3 = createAssetNode(test, "item3");
+        createPredictedTag(predicted3, "red", 0.5);
+        predicted3.setProperty("x", "red");
+
+        root.commit();
+
+        assertQuery("//element(*, dam:Asset)[jcr:contains(., 'red')]", XPATH, Arrays.asList("/test/item1", "/test/item2", "/test/item3"));
+        assertOrderedQuery("select [jcr:path] from [dam:Asset] where contains(*, 'red')", Arrays.asList("/test/item2", "/test/item3", "/test/item1"));
```

Comment:
Thomas, do you have idea why this failed, it don't return proper order per confidence